extensions/prototype-content-with-issues-table/app.R

```diff
@@ -0,0 +1,197 @@
+    if (nrow(last_visit) == 0){
+      visit_timestamp <- as.POSIXct(0) 
+      visitor_username <- "No visit data exists for this content"
+    } else {
+      visit_timestamp <- last_visit$timestamp
+      visitor_username <- get_user_info(client, last_visit$user_guid, connectServer)
+    }
+    if (is.na(item$content$title)) {
+      item$content$title <- "" # use empty strings when content is missing title
+    }
+    # return required information from https://github.com/posit-dev/connect/issues/30288 
+    all_failed_jobs <- bind_rows(lapply(seq_len(nrow(failed_jobs)), function(i) {
+      tibble(
+        "content_title" = item$content$title,
+        "content_guid" = item$content$guid,
+        "content_owner" = item$content$owner[[1]]$username,
+        "job_failed_at" = failed_jobs$end_time[i],
+        "failed_job_type" = failed_jobs$tag[i],
+        "failure_reason" = failed_jobs$exit_code[i],
+        "last_deployed_time" = item$content$last_deployed_time,
+        "last_viewed_by" = visitor_username,
+        "last_viewed_time" = visit_timestamp
+      )
+    }))
+    print(all_failed_jobs)
+    all_failed_jobs
+  }
+}
+
+server <- function(input, output, session) {
+  # use API key environment variables for now
+  # we can add local run and connect API integration functionality later
+  api_key <- Sys.getenv("CONNECT_API_KEY")
+  connect_server <- Sys.getenv("CONNECT_SERVER")
+  # initialize Connect API client
+  client <- connect(server = connect_server, api_key = api_key)
+  # set dashboard URL, needed for call to v1/users for visitor user_guid replacement
+  connectServer <- client$get_dashboard_url()
+  # get content once up front and pass it around for additional filtering
+  content <- get_content(client, limit = inf)
+  
+  # cache content list
+  content_list <- reactive({
+    as_content_list(content, client)
+  }) |> bindCache("static_key")
+  
+  # cache usage (uses firehose if available, legacy otherwise)
+  usage <- reactive({
+    get_usage(client)
+  }) |> bindCache("static_key")
+
+  # cache content with failed jobs 
+  bad_content_df <- reactive({
+    req(content_list(), usage())
+    map_dfr(content_list(), ~ get_failed_job_data(.x, usage(), client, connectServer))
+  }) |> bindCache("static_key")
+  
+  # output the table on load, wrapping in render UI 
+  output$jobs <- renderDT({
+    datatable(bad_content_df(), 
+              rownames = FALSE, 
+              escape = FALSE,
+              options = list( # non-interactive table for this prototype
+                paging = FALSE,
+                searching = FALSE,
+                ordering = FALSE, 
+                info = FALSE, 
+                dom = "t" 
+              )
+            )
+  })
+}
+
+ui <- fluidPage(
+  tags$head(
+    # icons for pretty display 
+    tags$link(rel = "stylesheet", href = "https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css")
+  ),
```

Comment:
Is this the right way to do something like this? I thought font awesome was bundled already? Also would `includeCSS()` be better than adding a tag to the head?

Reply:
And, also, presentation layer like fonts like this is a premature optimization at this point, no?

Reply:
Noted your feedback on bundled FA and `includeCSS()`, and will remove lingering styling.